flake.nix

```diff
@@ -187,29 +187,39 @@
           # Customize source filtering as Nickel uses non-standard-Rust files like `*.lalrpop`.
           src = filterNickelSrc craneLib.filterCargoSources;
 
-          # Args passed to all `cargo` invocations by Crane.
-          cargoExtraArgs = "--frozen --offline --workspace";
-        in
-        rec {
+          # set of cargo args common to all builds
+          cargoBuildExtraArgs = "--frozen --offline";
+
           # Build *just* the cargo dependencies, so we can reuse all of that work (e.g. via cachix) when running in CI
           cargoArtifacts = craneLib.buildDepsOnly {
-            inherit
-              src
-              cargoExtraArgs;
+            inherit src;
+            cargoExtraArgs = "${cargoBuildExtraArgs} --workspace";
+            buildInputs = [ pkgs.python3 ];
           };
 
-          nickel = craneLib.buildPackage {
-            inherit
-              src
-              cargoExtraArgs
-              cargoArtifacts;
-          };
+          buildPackage = packageName:
+            let
+            in
+            craneLib.buildPackage {
+              inherit
+                src
+                cargoArtifacts;
+
+              cargoExtraArgs = "${cargoBuildExtraArgs} --package ${packageName}";
+            };
+
+
+        in
+        rec {
+          nickel = buildPackage "nickel-lang";
+          lsp-nls = buildPackage "nickel-lang-lsp";
+          utilities = buildPackage "nickel-lang-utilities";
+          nickel-wasm-repl = buildPackage "nickel-repl";
 
           rustfmt = craneLib.cargoFmt {
             # Notice that unlike other Crane derivations, we do not pass `cargoArtifacts` to `cargoFmt`, because it does not need access to dependencies to format the code.
             inherit src;
 
-            # We don't reuse the `cargoExtraArgs` in scope because `cargo fmt` does not accept nor need any of `--frozen`, `--offline` or `--workspace`
```

Comment:
Is this deletion on purpose?

Reply:
Yes, I've changed how `cargoExtraArgs`s are handled in the file